### ISSUE TITLE
Using multiple providers

### ISSUE THREAD

**Issue by kthchew** [mholt/caddy-dynamicdns#47]
Is it possible to use multiple providers to update different domains? I tried this config (simplified)

```
{
    dynamic_dns {
        provider cloudflare {$CLOUDFLARE_API_TOKEN}
        domains {
            main-domain.com
        }
    }

    dynamic_dns {
        provider duckdns {$DUCKDNS_TOKEN}
        domains {
            secondary-domain.com
        }
    }
}
```

but it seems like only the second `dynamic_dns` block (for DuckDNS) is functional; `main-domain.com` is not updated at all.

**Comment by francislavoie** [comment]
I don't think it is possible right now unfortunately. It would require a restructure of the module config to handle that 😓

As a workaround you could run two instances of Caddy, one actually serving HTTP + ddns for one provider, and the other just taking care of ddns for the other provider with no site blocks.

**Comment by mholt** [comment]
Huh, interesting. You want two domains from different providers pointed to your machine?

Haven't seen that need yet -- but if it's legitimate we can probably refactor some things.

**Comment by mholt** [comment]
Awesome, good to know. Thanks!

I'll close this for now but we can revisit later if there's a more compelling use case for it.

**Comment by maxbec** [comment]
I also would like to see this option. I have multiple services running on my server. Some for my private network and some for my little company. That's why i need two domains.

**Comment by mholt** [comment]
This might be a big change but if someone wants to discuss the changes and submit a PR, I'd be okay with reviewing/discussing it!

**Comment by Monviech** [comment]
I'm interested in this since I get asked often if its possible to implement this in the opnsense plugin. There seems to be a use-case for some caddy users.

I can not contribute it myself though.

**Comment by mietzen** [comment]
I would also have another use case for this, I created a [OPNsense DNS Module](https://github.com/mietzen/caddy-dns-opnsense):

		provider opnsense {
			host {env.OPNSENSE_HOSTNAME}
			api_key {env.OPNSENSE_API_KEY}
			api_secret_key {env.OPNSENSE_API_SECRET}
			dns_service dnsmasq # or unbound
			insecure # Optional: skip TLS verification for self-signed certs
			entry_description Managed by Caddy # Optional
		}

I use caddy-dynamicdns with that to set local DNS entries in OPNsense unbound / dnsmasq.

It would be nice if we could use a 2nd instance for DDNS domains.
